testdata/p4_16_samples_outputs/pna-lookahead-structure-bit-field-midend.p4

```diff
@@ -21,9 +21,11 @@ struct headers_t {
 }
 
 parser MainParserImpl(packet_in pkt, out headers_t hdr, inout main_metadata_t main_meta, in pna_main_parser_input_metadata_t istd) {
+    bit<8> tmp;
     state start {
-        main_meta.f1 = pkt.lookahead<bit<8>>();
-        transition select(main_meta.f1) {
+        tmp = pkt.lookahead<bit<8>>();
+        main_meta.f1 = tmp;
+        transition select(tmp) {
```

Comment:
Option 2 from https://github.com/p4lang/p4c/issues/4194#issuecomment-1764870103 is the main motivation here.

Currently, `ExpandLookahead` doesn't apply to `bit<8>` for example. Then `non_field_lhs = pkt.lookahead<bit<8>>()` will fail on BMv2, because it will be transformed to use `assign`, which doesn't support lookahead. After the change, it becomes `tmp = pkt.lookahead<bit<8>>(); non_field_lhs = tmp`. This shall work because `tmp = pkt.lookahead<bit<8>>()` will be transformed to use `set` that does support lookahead.

Reply:
This solution looks brittle, because inserting a new copy propagation pass in between ExpandLookahead and the backend may remove the temporary.
I think a more robust solution would actually insert the temporary just before the final code generation.

Reply:
Maybe moving or adding another ExpandLookahead towards the end of the midend pass manager in https://github.com/p4lang/p4c/blob/main/backends/bmv2/simple_switch/midend.cpp could work?